File: solutions/elixir/eliuds-eggs/1/lib/eliuds_eggs.ex
defmodule EliudsEggs do
  @moduledoc false

  @doc """
  Given the number, count the number of eggs.
  """
  @spec egg_count(number :: integer()) :: non_neg_integer()
  def egg_count(0), do: 0
  def egg_count(n), do: rem(n, 2) + egg_count(div(n, 2))
end

File: solutions/kotlin/eliuds-eggs/1/src/main/kotlin/EliudsEggs.kt
object EliudsEggs {

    fun eggCount(number: Int): Int{
        tailrec fun go(n: Int, eggs: Int): Int {
            return if (n == 0) eggs else go(n / 2, (n and 1) + eggs)
        }

        require(number >= 0)
        return go(number, 0)
    }
}
